### DISCUSSION TITLE
Re DatabaseSeeder.php

### DISCUSSION THREAD

**Discussion by samcoppock** [wintercms/winter#1178] [Q&A] [answered]
the file
'modules/backend/database/seeds/DatabaseSeeder.php'

has this code in it

```
$shouldRandomizePassword = SeedSetupAdmin::$password === '';
$adminPassword = $shouldRandomizePassword ? Str::random(22) : SeedSetupAdmin::$password;
```

which seams to enable you to set the value of 

modules/backend/database/seeds/SeedSetupAdmin.php
`public static $password = '';`

from outside of the class.

Does the fact that it checks that the password is set before using a random string or the one that has been set ( if it has been set ) mean that setting the value of password from my own code ( using SeedSetupAdmin::$password == 'whatever' ) is something that i am intended to be able to do and will be safe against future winter updates?

**Comment by bennothommo** [comment, marked as answer]
@samcoppock yes, it was intended to allow people who have specialised setups (or processes) and wanted to be able to ensure a specific password when migrating for the first time.

We have no plans to change this, so you can feel free to use it for your purposes.
